addons/test_website/static/tests/tours/page_manager.js

```diff
@@ -0,0 +1,58 @@
+/** @odoo-module **/
+
+import wTourUtils from 'website.tour_utils';
+
+wTourUtils.registerWebsitePreviewTour('test_website_page_manager', {
+    test: true,
+    url: '/',
+}, [
+// Part 1 of the test: check that the website filter is working
+{
+    content: "Click on Site",
+    trigger: 'button.dropdown-toggle[title="Site"]',
+}, {
+    content: "Click on Test Models",
+    trigger: 'a.dropdown-item[data-menu-xmlid="test_website.menu_test_model_pages"]',
+}, {
+    content: "Check that we see records from My Website",
+    trigger: ".o_list_table .o_data_row .o_data_cell[name=name]:contains('Test Model Website 1') " +
+             "~ .o_data_cell[name=website_id]:contains('My Website')",
+    run: () => null, // it's a check
+}, {
+    content: "Check that there is only 2 records in the pager",
+    trigger: ".o_pager .o_pager_value:contains('1-2')",
+    run: () => null, // it's a check
+}, {
+    content: "Click on the 'Select all records' checkbox",
+    trigger: "thead .o_list_record_selector",
+}, {
+    content: "Check that there is only 2 records selected",
+    trigger: ".o_list_selection_box:contains('2 selected')",
+    run: () => null, // it's a check
+}, {
+    content: "Click on My Website search filter",
+    trigger: "button.dropdown-toggle:contains('My Website')",
+}, {
+    content: "Select My Website 2",
+    trigger: ".dropdown-menu.show > .dropdown-item:contains('My Website 2')",
+}, {
+    // This step is just here to ensure there is more records than the 2
+    // available on website 1, to ensure the test is actually testing something.
+    content: "Check that we see records from My Website 2",
+    trigger: ".o_list_table .o_data_row .o_data_cell[name=name]:contains('Test Model Website 2') " +
+             "~ .o_data_cell[name=website_id]:contains('My Website 2')",
+    run: () => null, // it's a check
+},
+// Part 2 of the test: ensure Kanban View is working / not crashing
+{
+    content: "Click on Kanban View",
+    trigger: '.o_cp_switch_buttons .o_kanban',
+}, {
+    content: "Click on List View",
+    extra_trigger: '.o_kanban_renderer',
+    trigger: '.o_cp_switch_buttons .o_list',
+}, {
+    content: "Wait for List View to be loaded",
+    trigger: '.o_list_renderer',
+    run: () => null, // it's a check
+}]);
```

Comment:
Note: This is not covering the case fixed by this PR, as the list view has `website_pages_list` js_class.
It's only there as a safeguard in case we would break it later even if `website_pages_list` is set.

Maybe we could purposely not set the `website_pages_list` on this test list view.. But then it would not test the regular case ^^.

Or now that I write it, maybe I could add 2 list view, with and without..